pkg/hashreader/hashreader.go

```diff
@@ -0,0 +1,120 @@
+package hashreader
```

Comment:
I think this package can be named `hasher` and having a public struct `Reader`

Reply:
Will keep it as hashtee instead hasher means closer to hash.Hash which is not the case here.